src/Fields/ArrayBlocks.jl

```diff
@@ -152,6 +152,18 @@ function evaluate!(cache,::typeof(unwrap_cached_array),a::ArrayBlock)
   r
 end
 
+Arrays.setsize!(a::ArrayBlock,s::NTuple{N,<:Integer}) where {N} = @notimplemented
+
+function Arrays.setsize!(a::ArrayBlock,b::ArrayBlock)
```

Comment:
I am afraid that this would not work when we have nested blocks. I have to think how to implement this.

Reply:
I see, I had not thought about nested blocks...

Reply:
It seems that are tests are about to pass with the new implementation